### ISSUE TITLE
Inform users that LOOT doesn't automatically sort plugins on launch

### ISSUE THREAD

**Issue by Nazenn** [loot/loot#528]
Since the new UI was introduced a few versions ago, there has been a huge influx of support requests on various modding communities in regards to issues with load order that are found to have originated at the start with confusion about the new LOOT.

Because LOOT now immediately displays a load order compared to the last version, people think just running it is enough and don't notice the Sort Plugins button. As a result a lot of people think they have run LOOT on their load order but don't realize it has more steps and as such aren't properly applying LOOTs changes or receiving LOOT's warnings.

On average I see approximately two to three reports per day of help issues where LOOT has not properly been used due to people not being able to easily find the sort button.

I propose that this be changed into text instead of a symbol as the symbol is too ambiguous and not easily recognizable, as well as a warning, perhaps that can be optionally turned off, that if people try and close the program without actually hitting Sort Plugins and Apply, it delivers a warning that they have not properly applied the changes to their load order and asks them to confirm that they do want to leave before sorting their load order.

I understand you can't account for everyone or everything, but right now it really is a major thing with people being adamant that they have run LOOT only to realize the real option is hidden away and they hadn't been running it all along.

**Comment by WrinklyNinja** [comment]
> Since the new UI was introduced a few versions ago, there has been a huge influx of support requests on various modding communities

You'd think that users would RTFM and seek support from the LOOT team directly (we're not exactly hard to contact), but no... :unamused: 

> I propose that this be changed into text instead of a symbol as the symbol is too ambiguous and not easily recognizable

I'll consider this, or some equivalent UI change to make the button more prominent.

> as well as a warning, perhaps that can be optionally turned off, that if people try and close the program without actually hitting Sort Plugins and Apply, it delivers a warning that they have not properly applied the changes to their load order and asks them to confirm that they do want to leave before sorting their load order.

If you've hit Sort Plugins but not applied or cancelled, LOOT will already prompt on exit. Having a warning if you haven't hit Sort Plugins though could be incredibly annoying. I'd be surprised if I was alone in opening LOOT to check for messages, etc. but not sorting. 

I don't like the idea of making UI behaviour optional, it introduces inconsistency in how users expect to interact with the application, eg. a user turning the prompt off but then forgetting they did that, and assuming that they're setting the load order because it isn't prompting them about it.

**Comment by Nazenn** [comment]
I think github scares a lot of people away honestly which is a shame, especially people who struggle even with the standard forum method of communication with dev teams. Also I think that because a lot of people don't realize they are doing it wrong, its just not something they think of reporting. Perhaps a "report issues" link on the main page (https://loot.github.io/) could help people realize that they can actually report this stuff? I didn't even know this section was here, being unfamiliar with github myself, until it was linked to me. I know when you click on the download button on that page, issues is one of the tabs, but I can honestly say I've never mentioned it to someone and actually had them notice it was there before I pointed it out.

Yeah I understand the issue with having optional warnings etc, I don't particularly like it either, it was just the first thing that came to mind when thinking about it from a new user point of view. Perhaps instead just changing the icon to say Sort Plugins and putting a message at the top when you open the program but haven't hit the button saying "This is your current load order display only, it has not been sorted by LOOT." or something along those lines together would help to fix the majority of the confusion.

**Comment by Freso** [comment]
> I think github scares a lot of people away honestly which is a shame,

One of the first messages always being displayed and in all the masterlists is a link to the official LOOT forum thread…
GitHub is "closer to the iron", sure, but reporting issues in that forum thread is just as valid, and lots of people are monitoring that thread that are not necessarily keeping a close eye on GitHub issues, so may even sometimes be a better bet at getting support.

> putting a message at the top when you open the program but haven't hit the button saying "This is your current load order display only, it has not been sorted by LOOT." or something along those lines

I actually like this idea. This allows people to just pop open LOOT and check load order messages, without having to be interrupted by a dialogue when exiting without sorting.

All that said, I definitely thinks it's a good idea to replace icon-only buttons with text-based ones or text/icon combinations. Pictographs mean different things to different people, and a pictograph's obvious meaning to you might be the opposite of what comes to someone else's mind.

**Comment by WrinklyNinja** [comment]
Having a general message displayed whenever a load order has not been sorted seems like a good idea to me too.

Putting text instead of / in addition to icons in buttons generally will add more UI clutter though, that's the advantage of tooltips. Maybe we need to do a sort of audit of which buttons could be unclear and need to be really obvious?

**Comment by Freso** [comment]
Related comment in unrelated forum: http://chatlogs.metabrainz.org/brainzbot/musicbrainz/msg/3476633/

**Comment by WrinklyNinja** [comment]
For the warning message saying that the user hasn't yet sorted their load order, can either or you think of a good wording that doesn't suggest that it's *never* been sorted by LOOT?

**Comment by Nazenn** [comment]
"You have not sorted your current load order with LOOT this session"?

**Comment by pStyl3** [comment]
I like the idea of telling the user, that s/he has not yet sorted his/her load order with LOOT, aswell. How about these alternative lines?

* `No initial load order sorting has been carried out.`
* `You have not carried out a first and initial sorting.`
* `You're load order has not yet undergo a first initial sorting. Please sort your load order.`

However, I think that the buttons 'Sort Plugins', 'Update Masterlist' and 'Search Cards' do not necessarily need to be changed into text equivalents .. it isn't that hard to move your cursor over a button and wait 1s to see whether or not a tooltip pops up which explains it's function .. that is not only applicable to LOOT, but every program and tool out there.

Not to mention, that available space is always rare. For [example](http://abload.de/img/lootvzuxz.jpg), if the LOOT window is only displayed on half width of the monitor, having the buttons displayed as pictograms is certainly freeing up some space in the header bar (and as this is a valid state the LOOT window can be in, it shouldn't be neglected).

All that said .. making the sort button more prominent should be certainly an option. As a general consensus seems to prevail that a warning should be added, if the load order never has been sorted by LOOT yet, how about the following: give that warning message, however it may turn out, a colored outline + display the sort pictogram in that color *aswell* .. until the sort button has been used. Could look like [this](http://abload.de/img/loot2bhaon.jpg) .. with or without the circle around it. Or, if the sort button should not be displayed permanently in another color (until the sort button has been used), another way could also be to let the sort button blink every, let's say, 5 seconds (any less would be probably to irritating).

That way, it could be more obvious that the user hasn't performed a crucial step yet (sorting),

**Comment by Freso** [comment]
> How about these alternative lines?

All those three lines don't really differentiate between LOOT having *ever* been used to sort and LOOT having not been used to sort during this "session" (which is the primary issue with the current wording).

> it isn't that hard to move your cursor over a button and wait 1s to see whether or not a tooltip pops up which explains it's function

It might not be hard for you, but for some people that *is* hard: http://chatlogs.metabrainz.org/brainzbot/musicbrainz/msg/3476634/
Also, waiting for a tooltip to pop up creates an artificial delay, even if it's "just" 1 second (probably slightly more, as you also have to read and comprehend the message popping up), it's still a second more than recognising the button and clicking it. This will become "learned behaviour" for most users, but some users will not learn as fast as others, in particular if you only use LOOT every few months.
Additionally, many users are not very "explorative". They don't just hover over random icon expecting that to tell them what the button does. (That is, if they even recognise that the icon is a button.) They will just stare blankly at the interface, confused and bewildered, not knowing what's going on or where to go from here. *Hopefully* that will cause them to check our the documentation, but…

> display the sort pictogram in that color *aswell* .. until the sort button has been used

I like this idea. 👍 (And I really, really dislike the blinking, even if it was every 5 hours. Please just don't. :disappointed:)

**Comment by pStyl3** [comment]
New alternative lines with a slightly different approch:
* `LOOT needs to reevaluate your load order. Please use the Sort button.`
* `LOOT should reevaluate your load order. Please use the Sort button.`

@Freso I will comment your reply later, for now I have a question for @WrinklyNinja:
Would it be possible to let LOOT display animated .gif's (like for example in a new window)? Or (unanimated) standard image files like .jpg to begin with?

**Comment by WrinklyNinja** [comment]
The conversation in this issue thread can be separated into a few separate issues, so I've changed the title for this issue to reflect the one that I've already started work on and referred to this issue as in commit messages.

How to better draw attention to the sorting button / clarify its meaning should be split off into another issue, and the conversation about icons vs. text into a third.

> New alternative lines with a slightly different approch:
> 
> LOOT needs to reevaluate your load order. Please use the Sort button.
> LOOT should reevaluate your load order. Please use the Sort button.

Those imply that LOOT can detect there is something wrong or different with the existing order, leading to possible confusion if LOOT then makes no changes.

> @WrinklyNinja:
> Would it be possible to let LOOT display animated .gif's (like for example in a new window)? Or to begin with, standard image files like e.g. .jpg ?

Yes, the UI is a Chromium browser window, so any file format it supports can be displayed.
